c/tskit/tables.c

```diff
@@ -7512,6 +7513,32 @@ ancestor_mapper_add_ancestry(ancestor_mapper_t *self, tsk_id_t input_id, double
     return ret;
 }
 
+static int
+ancestor_mapper_oldest_node(ancestor_mapper_t *self)
+{
+    const double *node_time = self->tables->nodes.time;
+    tsk_size_t a;
+    int ret = 0;
+    double max_time = -1;
+
+    for (a = 0; a < self->num_ancestors; a++) {
+        if (node_time[self->ancestors[a]] > max_time) {
+            max_time = node_time[self->ancestors[a]];
+        }
+    }
+    for (a = 0; a < self->num_samples; a++) {
+        if (node_time[self->samples[a]] > max_time) {
+            max_time = node_time[self->samples[a]];
+        }
+    }
+    if (max_time < 0) {
```

Comment:
I guess this can only happen if we have zero samples and zero ancestors, which we're already checking for?